### PR TITLE
[all] Externalize cxxopts

### PR DESCRIPTION
**Pull request by olivier-roussel**
This PR fixes a requirement of https://github.com/conda-forge/staged-recipes/pull/23085 for the SOFA conda package on conda-forge.
This add a find package to the cmakelists to avoid the cmake fetch of cxxopts sources if a cxxopts is already available and found. This is a simplified version of a find or fetch mechanism which looks enough here, and does not require cmake >= 3.24 (which embedds a more sophisticated find or fetch within an improved FetchContent set of macros).
The PR also adds a cmake SOFA option, which may be re-used if other dependencies will be managed by a similar mechanism, to allow or disable the fetch content of cmake (default behavior is to allow fetching to keep compatibility). 
This is required by the conda-forge PR to disable the fetching of source.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Review by fredroy — APPROVED** [review]
Could be nice to describe the list of the  fetched deps in the comment (or an other visible place)
 Otherwise the user will think it will fetch all dependencies (boost, eigen, qt... etc)

**Comment by olivier-roussel** [comment]
> Could be nice to describe the list of the fetched deps in the comment (or an other visible place) Otherwise the user will think it will fetch all dependencies (boost, eigen, qt... etc)

Agreed. Done.

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,6 +34,11 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${LIBRARY_OUTPUT_DIRECTOR
 # Option for packaging
 option(SOFA_BUILD_RELEASE_PACKAGE "Run package specific configure" OFF)
 
+# Option to allow some dependencies such as cxxopts to be fetched by cmake if
+# the package is not found
+option(SOFA_ALLOW_FETCH_DEPENDENCIES "Allow compatible dependencies to be fetched is the package is not found by cmake.
```
Comment (epernod) [inline CMakeLists.txt]:
```suggestion
option(SOFA_ALLOW_FETCH_DEPENDENCIES "Allow compatible dependencies to be fetched if the package is not found by cmake.
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -34,6 +34,11 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${LIBRARY_OUTPUT_DIRECTOR
...
+  Fetchable dependencies: cxxopts" ON)
```
Comment (epernod) [inline CMakeLists.txt]:
```suggestion
  List of dependencies that can be fetched: cxxopts" ON)
```

**Review by epernod — COMMENTED** [review]
small typo suggestions

**Pull request merged**